### DISCUSSION TITLE
How to implement dynamic mixing(DM) with Sepformer and where is the code implementation?

### DISCUSSION THREAD

**Discussion by Jasonare** [speechbrain/speechbrain#1672] [Q&A] [answered]
Hi, Thanks for your code work, it's much helpful for me! 
I have already trained the original sepformer and obtain 20.4 SI-SNRi, which is similar with in the paper.
Now, I wanna train sepformer+dm, but I don't find the code implementation of DM, especially how to dynamic mix the speech under mypath/wsj0/si_tr_s regularly, can you tell me where it is, please?
Besides, I see the training scipt of dynamic mixing you provided, it looks fairly simple, do I need to change anything else?

**Comment by ycemsubakan** [comment, marked as answer]
Hi, sorry for my very late reply.

Basically if you run the code with `python train.py hparams/sepformer.yaml --data_folder yourpath/wsj0-mix/2speakers --base_folder_dm yourpath/wsj0/si_tr_s --dynamic_mixing True` It will automatically create the resampled (if necessary) `wsj0` folder, and train your model with dynamic mixing. (You need to have the `wsj0` folder with .wav files)
